client/app/core/authorization.run.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-useless-return */
+/** @ngInject */
+
+export function authInit ($rootScope, $state, $log, Session, $sessionStorage, Language, ApplianceInfo, $window, RBAC, Redux, $ngRedux) {
+  $rootScope.$on('$stateChangeStart', changeStart)
+  $rootScope.$on('$stateChangeError', changeError)
+  $rootScope.$on('$stateChangeSuccess', changeSuccess)
+
+  $sessionStorage.$sync()  // needed when called right on reload
+  if ($sessionStorage.token) {
+    syncSession()
+  }
+
+  function changeStart (event, toState, toParams, _fromState, _fromParams) {
+    Redux.loadRedux().then(() => {
+      // first lets see if they are logged in and session is valid
+      event.preventDefault()
+      let nextState = ''
+      if (!Session.active()) {
+        nextState = 'login'
+        return
+      }
+      const isAuthorized = authorizedForState(toState) // assumes they are logged in and authorized to see the page
+      if (!isAuthorized) {
+        nextState = 'dashboard'
+        // transition them to login if the state they trying to go to is login
+      }
+      if (nextState) {
+        $state.transitionTo(nextState)
+        return
+      } else {
+        return
+      }
+    })
+  }
+  function authorizedForState (toState) {
+    let authorized = true
+    if (angular.isDefined(toState.data)) {
```

Comment:
why not use the redux state? (I think I'm missing the bigger play here)

Reply:
I would have to sit down and look at redux state but I think redux state might be out of context for this change.  State in this case refers to an Angular UI router state.  This project uses angular UI router.